speclang_modern.tex

```diff
@@ -2108,6 +2099,40 @@ \subsection{Non-terminating functions}
 to always terminate, which is the expected behavior of most
 functions.
 
+The \lstinline|terminates| clause can also be specified in a statement contract
+with the same meaning. When such a clause is not specified, it inherits the
+one associated to the innermost block containing it (recursively, until a
+block with a \lstinline|terminates| clause is found or the body of the function
+is reached, and then the \lstinline|terminates| clause of the function applies).
+When a statement \lstinline|S| is specified with a \lstinline|terminates| clause
+\lstinline|ST|:
+
+\begin{itemize}
+  \item the termination clause \lstinline|ET| of the innermost block containing
+        it must imply \lstinline|ST|,
+  \item statements of \lstinline|S| must terminate if \lstinline|ST| holds, but
+        the termination clause \lstinline|ET| of any block containing
+        \lstinline|S| is ignored.
```

Comment:
I don't understand the part after the word *but*. Why do you need to write something about blocks? In other word, could you explain why these 3 lines cannot be replaced by:
-   \item statements of \lstinline|S| must terminate if their inherited termination condition  \lstinline|ST| holds
        (the termination condition \lstinline|ET| is ignored for this verification).

Reply:
Changed